agent/agent.go

```diff
@@ -209,6 +256,21 @@ func (a *Agent) registerConsulService() error {
 	return nil
 }
 
+func (a *Agent) startRegisterLoop() error {
+	attempts := 0
+	for {
+		err := a.registerConsulService()
+		if (err != nil) && (attempts < maxConsulRegisterRetries) {
+			log.Println("An error ocurred while attempting to register:", err)
+			time.Sleep(3 * time.Second)
+			attempts++
+			continue
+		}
+
+		return err
+	}
+}
+
```

Comment:
as @arbulu89 pointed out, we should instead wait for the consul agent startup and then call `registerConsulService()` without retries.

Reply:
@stefanotorresi the thing with that approach is that `trento` will attempt to register with the consul instance too quick (before it's actually ready). It will fail and immediately quit. The previous approach was just sleeping for a few seconds, but that is a very ugly workaround.

We could change indeed the order so that the embedded consul instance starts first (it makes more logical sense), but I don't think it will remove the need of the register loop. Initially I was hoping I could somehow signal trento from the embedded consul waitgroup, but doing so requires access to some private attributes / functions from consul that are we can't use.